internal/log/log.go

```diff
@@ -17,110 +17,96 @@ type Logger interface {
 }
 
 func init() {
-	defaultLogger = &stdLogger
+	logger = standard
+	SetLogLevel(INFO)
 }
 
+type Level int
+
 // Allows the ordering of severity to be checked
 const (
-	FATAL = 60
-	ERROR = 50
-	WARN  = 40
-	INFO  = 30
-	DEBUG = 20
-	TRACE = 10
+	TRACE = Level(iota - 2)
+	DEBUG
+	INFO
+	WARN
+	ERROR
+	FATAL
 )
 
 var (
-	defaultLogger Logger
-	logLevel      int = INFO
-	logLock       sync.Mutex
-	IsError       bool = true
-	IsWarn        bool = true
-	IsInfo        bool = true
-	IsDebug       bool = false
-	IsTrace       bool = false
+	logger Interface
+	level  Level
+	lock   sync.Mutex
+	// FATAL level is never disabled
+	IsError bool
+	IsWarn  bool
+	IsInfo  bool
+	IsDebug bool
+	IsTrace bool
 )
 
-func SetLogLevel(level int) {
-	logLock.Lock()
+func SetLogLevel(lvl Level) {
+	lock.Lock()
+	IsTrace = false
+	IsDebug = false
+	IsInfo = false
+	IsWarn = false
+	IsError = false
+	IsTrace = false
+	if lvl != TRACE && lvl != DEBUG && lvl != INFO && lvl != WARN && lvl != ERROR && lvl != FATAL {
+		lvl = INFO
+	}
+	level = lvl
 	switch {
```

Comment:
Why not just turn the `switch` to:
```
switch level {
   case TRACE:
     …
  case DEBUG:
   …
}
```
I think that reads better.
Since with the fallthrough, understands it's going to the include the next set of cases.